woocommerce/woocommerce-android
# Issue/6119 refactor fetch variations

Partially fixes #6119


[FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2364) needs to be merged first and the fluxc hash updated in this PR.

### Description


This PR refactors fetchProductVariations into a suspendable function - removes usage of the event bus.

There are two reasons for this change

1. [The event bus architecture is considered as legacy/deprecated](https://github.com/wordpress-mobile/WordPress-FluxC-Android/wiki/%5BDeprecated%5D-Architecture)
2. WCAndroid app uses stateful repositories which need to be registered to the event bus - this leads to memory leaks when the repository doesn't unregister from the event bus. It also causes side-effects when multiple instances of a repository are created - eg. duplicate tracking events.


### Testing instructions

1. Open Variation Product detail
2. Open variation list
3. Notice that all variations are loaded
4. Add a new variation in WPAdmin
5. Repeat steps 1-3 and make sure the new variation appears
------
1. Add a new variation in WPAdmin (make sure the price is not empty)
2. Start the order creation flow
3. Add a product with variations
4. Make sure the variation created in step 1 is there
